source/geod24/LocalRest.d

```diff
@@ -182,105 +148,268 @@ private enum Status
 /// Data sent by the callee back to the caller
 private struct Response
 {
+    /// Response id
+    size_t id;
     /// Final status of a request (failed, timeout, success, etc)
     Status status;
-    /// ID used for request re-entrancy (See ID definition)
-    ID id;
     /// If `status == Status.Success`, the serialized return value.
     /// Otherwise, it contains `Exception.toString()`.
     SerializedData data;
 }
 
-/// Thrown when the sent request is faulty (e.g. 4xx HTTP error types)
-public class ClientException : Exception
+/// Ask the node to shut down
+private struct ShutdownCommand
 {
-    /// Constructor
-    public this (string msg,
-        string file = __FILE__, int line = __LINE__, Exception next = null)
-        @safe pure nothrow
-    {
-        super(msg, file, line, next);
-    }
+    /// Any callback to call before the Node's destructor is called
+    void function (Object) callback;
+
+    /// Whether we're restarting or really shutting down
+    bool restart;
 }
 
-/// Simple wrapper to deal with tuples
-/// Vibe.d might emit a pragma(msg) when T.length == 0
-private struct ArgWrapper (T...)
+/// Data sent by the caller
+private struct Command
 {
-    static if (T.length == 0)
-        size_t dummy;
-    T args;
+    /// Sequence id of Command
+    size_t id;
+    /// Method to call
+    string method;
+    /// Serialized arguments to the method
+    SerializedData args;
```

Comment:
Can you move them back to where  they were so they don't show as diff ?

Reply:
Once we squash the two LocalRest.d adaptation commits, they should be gone. I wanted to keep the two version in the history.